tss-esapi/tests/integration_tests/structures_tests/lists_tests/algorithm_property_list_tests.rs

```diff
@@ -0,0 +1,145 @@
+    let expected_algorithm_properties = vec![
+        (AlgorithmIdentifier::Rsa, AlgorithmAttributes(1)),
+        (AlgorithmIdentifier::Aes, AlgorithmAttributes(2)),
+    ];
+
+    let expected_tpml_alg_property = expected_algorithm_properties.iter().fold(
+        TPML_ALG_PROPERTY::default(),
+        |mut acc, (aid, aa)| {
+            acc.algProperties[acc.count as usize].alg = (*aid).into();
+            acc.algProperties[acc.count as usize].algProperties = (*aa).into();
+            acc.count += 1;
+            acc
+        },
+    );
+
+    let actual_algorithm_property_list: AlgorithmPropertyList = expected_tpml_alg_property
+        .try_into()
+        .expect("Failed to convert TPML_ALG_PROPERTY into AlgorithmPropertyList");
+
+    assert_eq!(
+        expected_algorithm_properties.len(), actual_algorithm_property_list.len(),
+        "The algorithm property list converted from TPML_ALG_PROPERTY did not contain the expected amount of elements"
+    );
+
+    expected_algorithm_properties
+        .iter()
+        .zip(actual_algorithm_property_list.iter())
+        .for_each(|((expected_aid, expected_aa), ap)| {
+            assert_eq!(
+                expected_aid, &ap.algorithm_identifier(),
+                "An AlgorithmProperty in the AlgorithmPropertyList converted from TPML_ALG_PROPERTY did not contain the expected algorithm identifer"
+            );
+            assert_eq!(
+                expected_aa, &ap.algorithm_properties(),
+                "An AlgorithmProperty in the AlgorithmPropertyList converted from TPML_ALG_PROPERTY did not contain the expected algorithm properties"
+            );
+        });
+
+    let actual_tpml_alg_property: TPML_ALG_PROPERTY = actual_algorithm_property_list.into();
+
+    crate::common::ensure_tpml_alg_property_equality(
+        &expected_tpml_alg_property,
+        &actual_tpml_alg_property,
+    );
+}
+
+#[test]
+fn test_valid_conversion_vector() {
+    let expected_algorithm_properties = vec![
+        (AlgorithmIdentifier::Rsa, AlgorithmAttributes(1)),
+        (AlgorithmIdentifier::Aes, AlgorithmAttributes(2)),
+    ];
+
+    let expected_tpml_alg_property = expected_algorithm_properties.iter().fold(
+        TPML_ALG_PROPERTY::default(),
+        |mut acc, (aid, aa)| {
+            acc.algProperties[acc.count as usize].alg = (*aid).into();
+            acc.algProperties[acc.count as usize].algProperties = (*aa).into();
+            acc.count += 1;
+            acc
+        },
+    );
+
+    let expected_algorithm_property_list: AlgorithmPropertyList = expected_tpml_alg_property
+        .try_into()
+        .expect("Failed to convert TPML_ALG_PROPERTY into AlgorithmPropertyList");
+
+    assert_eq!(
+        expected_algorithm_properties.len(), expected_algorithm_property_list.len(),
+        "The algorithm property list converted from TPML_ALG_PROPERTY did not contain the expected amount of elements"
+    );
+
+    let algorithm_properties_vector: Vec<AlgorithmProperty> =
```

Comment:
Usually `Vec<_>` is sufficient for type inference...

Reply:
Nice, to know. I like being explicit though unless I am feeling lazy.